src/flowgger/utils/rotating_file.rs

```diff
@@ -29,42 +62,96 @@ impl RotatingFile {
+    ///             https://docs.rs/chrono/0.3.1/chrono/format/strftime/index.html
     ///
     /// # Example
     /// From parameters:
     ///     - basename = 'logs/syslog.log'
+    ///     - max_size = 1024
     ///     - max_files = 2
     ///
     /// The following files will be generated:
-    ///     - Current file = 'logs/syslog.log'
-    ///     - Older file = 'logs/syslog.log.0'
-    ///     - Oldest file = 'logs/syslog.log.1'
+    ///     - Current file = 'logs/syslog.log', lg <= 1024
+    ///     - Older file = 'logs/syslog.log.0', lg <= 1024
+    ///     - Oldest file = 'logs/syslog.log.1', lg <= 1024
     ///
-    pub fn new<P: AsRef<Path>>(basepath: P, max_size: usize, max_files: i32) -> Self {
+    /// From parameters:
+    ///     - basename = 'logs/syslog.log'
+    ///     - max_time = 2
+    ///     - app started on 2018-01-08 at 01:43 UTC
+    ///     - time format is "%Y%m%dT%H%M%SZ"
+    ///
+    /// The following files will be generated:
+    ///     - Current file = 'logs/syslog-20180108T014343Z.log'
+    ///     - Older file = 'logs/syslog-20180108T014543Z.log'
+    ///     - Oldest file = 'logs/syslog-20180108T014743Z.log'
+    ///
+    pub fn new<P: AsRef<Path>>(basepath: P, max_size: usize, max_time: u32, max_files: i32, time_format: &str) -> Self {
         let basename = basepath.as_ref().to_path_buf();
         Self {
             basename,
             max_size,
+            max_time,
             max_files,
-
+            time_format: time_format.to_string(),
             current_file: None,
             current_size: 0,
+            next_rotation_time: None,
+
+            #[cfg(test)]
+            now_time_mock: Utc::now(),
         }
     }
 
+    fn get_current_date_time(&self) -> DateTime<Utc> {
+        #[cfg(test)]
+        return self.now_time_mock;
+
+        #[cfg(not(test))]
+        Utc::now()
+    }
+
+    /// Build an output file name appending the current timestamp, and compute the file expiration time
+    fn build_timestamped_filename(&mut self) -> PathBuf {
+        let current_time  = self.get_current_date_time();
+        println!("time got {}", current_time);
+        self.next_rotation_time = Some(current_time + Duration::minutes(i64::from(self.max_time)));
+
+        let dt_str = current_time.format(&self.time_format).to_string();
+        let mut new_file = self.basename.clone();
+        new_file.set_file_name(&format!("{}-{}.{}",
+                                self.basename.file_stem().unwrap_or_else(|| OsStr::new("")).to_string_lossy(),
+                                dt_str,
+                                self.basename.extension().unwrap_or_else(|| OsStr::new("")).to_string_lossy()));
+        new_file
+    }
+
     /// Open the base file and ready for logging and set it as current file
     /// Should typically be called after object creation in order to be able to start writing
+    /// The file opened is the basename specified at object creation.
```

Comment:
This section is not too clear can you reword it a bit?
Something like:

```
It creates a file, if it doesn't exist, and open for writing at the path pointed to in the basename field of the object
```
I'm struggling to put it into clearer wording myself really

Reply:
reworded, added examples